cloudify_files/__init__.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2017 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import grp
+import os
+import pwd
+import subprocess
+
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+
+
+def execute_command(_command, extra_args=None):
+
+    ctx.logger.debug('_command {0}.'.format(_command))
+
+    subprocess_args = {
+        'args': _command.split(),
+        'stdout': subprocess.PIPE,
+        'stderr': subprocess.PIPE
+    }
+    if extra_args is not None and isinstance(extra_args, dict):
+        subprocess_args.update(extra_args)
+
+    ctx.logger.debug('subprocess_args {0}.'.format(subprocess_args))
+
+    process = subprocess.Popen(**subprocess_args)
+    output, error = process.communicate()
+
+    ctx.logger.debug('command: {0} '.format(_command))
+    ctx.logger.debug('output: {0} '.format(output))
+    ctx.logger.debug('error: {0} '.format(error))
+    ctx.logger.debug('process.returncode: {0} '.format(process.returncode))
+
+    if process.returncode:
+        ctx.logger.error('Running `{0}` returns error.'.format(_command))
+        return False
+
+    return output
+
+
+class CloudifyFile(object):
+
+    def __init__(self, operation_inputs):
+        self.config = self.get_config(operation_inputs)
+        self.resource_path = self.config.get('resource_path')
+        self.file_path = self.config.get('file_path')
+        self.owner = self.config.get('owner')
+        self.mode = self.config.get('mode')
+        self.use_sudo = self.config.get('use_sudo')
+
+    @staticmethod
+    def get_config(inputs):
+        config = ctx.node.properties.get('resource_config', {})
+        config.update(
+            ctx.instance.runtime_properties.get('resource_config', {}))
+        config.update(inputs.get('resource_config', {}))
+        return config
+
+    def create(self):
+
+        downloaded_file_path = ctx.download_resource(self.resource_path)
+
+        if self.use_sudo:
+            cp_out = execute_command('sudo cp {0} {1}'.format(
```

Comment:
What do you think about use repr for such cases: "cp {0} {1}".format(repr("aa vvv"), repr("cc dd"))